src/features/user/MFV2/ForestProgress/index.tsx

```diff
@@ -0,0 +1,124 @@
+import { useState, useMemo, useEffect } from 'react';
+import styles from './ForestProgress.module.scss';
+import TargetsModal from './TargetsModal';
+import { useMyForestV2 } from '../../../common/Layout/MyForestContextV2';
+import EmptyProgress from './EmptyProgress';
+import ForestProgressItem from './ForestProgressItem';
+
+interface ProgressBarsProps {
+  handleEditTargets: () => void;
+  treesDonated: number;
+  areaRestored: number;
+  areaConserved: number;
+}
+
+const ProgressBars = ({
+  handleEditTargets,
+  treesDonated,
+  areaRestored,
+  areaConserved,
+}: ProgressBarsProps) => {
+  const { contributionsResult, treeTarget, restoreTarget, conservTarget } =
+    useMyForestV2();
+
+  const shouldShowBar = (target: number, contributedUnits: number): boolean =>
+    contributedUnits > 0 || target > 0;
+
+  return (
+    <>
+      {shouldShowBar(treeTarget, treesDonated) && (
+        <ForestProgressItem
+          handleEditTargets={handleEditTargets}
+          dataType={'treesPlanted'}
+          target={treeTarget}
+          gift={contributionsResult?.stats.treesDonated.received ?? 0}
+          personal={contributionsResult?.stats.treesDonated.personal ?? 0}
+        />
+      )}
+      {shouldShowBar(restoreTarget, areaRestored) && (
+        <ForestProgressItem
+          handleEditTargets={handleEditTargets}
+          dataType={'areaRestored'}
+          target={restoreTarget}
+          gift={contributionsResult?.stats.areaRestoredInM2.received ?? 0}
+          personal={contributionsResult?.stats.areaRestoredInM2.personal ?? 0}
+        />
+      )}
+      {shouldShowBar(conservTarget, areaConserved) && (
+        <ForestProgressItem
+          handleEditTargets={handleEditTargets}
+          dataType={'areaConserved'}
+          target={conservTarget}
+          gift={contributionsResult?.stats.areaConservedInM2.received ?? 0}
+          personal={contributionsResult?.stats.areaConservedInM2.personal ?? 0}
+        />
+      )}
+    </>
+  );
+};
+
+const ForestProgress = () => {
+  const [isEditingTargets, setIsEditingTargets] = useState(false);
+  const handleEditTargets = () => setIsEditingTargets(true);
+
+  const [isProgressEnabled, setIsProgressEnabled] = useState(false);
+  const { treeTarget, conservTarget, restoreTarget, contributionsResult } =
+    useMyForestV2();
+
+  const aggregateContributions = () => {
+    const treesDonated =
+      (contributionsResult?.stats.treesDonated.personal ?? 0) +
+      (contributionsResult?.stats.treesDonated.received ?? 0) +
+      (contributionsResult?.stats.treesRegistered ?? 0);
+    const areaRestored =
+      (contributionsResult?.stats.areaRestoredInM2.personal ?? 0) +
+      (contributionsResult?.stats.areaRestoredInM2.received ?? 0);
+    const areaConserved =
+      (contributionsResult?.stats.areaConservedInM2.personal ?? 0) +
+      (contributionsResult?.stats.areaConservedInM2.received ?? 0);
+    return { treesDonated, areaRestored, areaConserved };
```

Comment:
I suggest separating the `stats` object in MyForestContextV2 and just using that here. It will make the code easier to read. 

It also makes sense to separate the main parts of the `ContributionsResponse`